tests: kernel: Move tests/kernel/profiling to tests/kernel/threads

ramakrishnapallala opened the pull request:
Move tests/kernel/profiling test to tests/kernel/threads folder.

Fixes #7966

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

nashif reviewed (changes requested):
thats not what the issue is about, did not asking for moving the test. I am asking for testing the k_thread_foreach functions on the API level as part of thread testing and not as part of getting stack information listed. Those are two different things

ramakrishnapallala commented:
@nashif , I will move the test_k_thread_foreach test to threads_api tests.

Commit:
tests: kernel: Move k_thread_foreach() API test to thread_apis test

Move k_thread_foreach() API test from tests/kernel/profiling to
tests/kernel/threads/thread_apis.

Fixes #7966

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

andrewboie commented on tests/kernel/threads/thread_apis/prj.conf:
```diff
@@ -1,4 +1,6 @@
 CONFIG_THREAD_MONITOR=y
-CONFIG_HEAP_MEM_POOL_SIZE=256
 CONFIG_THREAD_CUSTOM_DATA=y
+CONFIG_INIT_STACKS=y
```
why is CONFIG_INIT_STACKS enabled? does the test use it?

ramakrishnapallala replied:
we don't need this in this test. I will remove.

nashif commented:
recheck

nashif commented:
recheck

nashif commented:
recheck
